api/maps/AddMarkers.tsx

```diff
@@ -92,15 +94,19 @@ export default function AddMarker({
     return mapZoom;
   };
 
-  const clusterer = useMemo(() => {
-    if (!map) return null;
+  useEffect(() => {
+    if (!map || !projects) return;
+
+    // clear previous cluster markers
+    if (clusterer) {
+      clusterer.clearMarkers();
+    }
```

Comment:
Your sprint instructions did not tell you to implement this functionality. 
This implementation re-renders the map each time, which makes an API call. So if I search 'abcdefg', I will be making 7 API calls which will become costly. 
As per my sprint assignments, Neha will be implementing this functionality in a way that will make less API calls and we will be merging her implementation.

Reply:
The code here is for updating clusters for filtering, not for re-rendering the map on search (but I fixed that now). If the clusters are not updated, then when filters are applied from the dropdown, the clusters from the markers that are not part of the filters remain on the map.

Reply:
I commented the exact line I am talking about. You are re-rendering the map when we filter, which we can't do because that is an API call. Please fix this.

Also, the updating clusters for filtering was also not part of your sprint.